internal/services/network/private_endpoint_resource.go

```diff
@@ -299,8 +300,17 @@ func resourcePrivateEndpointCreate(d *pluginsdk.ResourceData, meta interface{})
 			return fmt.Errorf("creating Private Endpoint %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 		}
 	}
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting for creation of Private Endpoint %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	err = pluginsdk.Retry(d.Timeout(pluginsdk.TimeoutCreate), func() *resource.RetryError {
+		if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+			return &resource.RetryError{
+				Err:       fmt.Errorf("waiting for creation of Private Endpoint %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err),
+				Retryable: strings.Contains(strings.ToLower(err.Error()), "retryable"),
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
```

Comment:
unfortunately this won't do anything since it's only retrying on the poller/future.. since this is an API error this'd need to be fixed here - is there a Swagger issue tracking this upstream issue?